pkg/monitor/monitorapi/types.go

```diff
@@ -197,6 +197,29 @@ func IsInNamespaces(namespaces sets.String) EventIntervalMatchesFunc {
 	}
 }
 
+// ContainsAllParts ensures that all listed key match at least one of the values.
+func ContainsAllParts(matchers map[string][]string) EventIntervalMatchesFunc {
+	return func(eventInterval EventInterval) bool {
+		actualParts := LocatorParts(eventInterval.Locator)
+		for key, possibleValues := range matchers {
+			actualValue := actualParts[key]
+
+			found := false
+			for _, possibleValue := range possibleValues {
```

Comment:
How about `strings.Contains(actualValue, possibleValue)`? this way I can filter for all pods of a certain pattern or maybe make it so we can pass a regex?